src/context.rs

```diff
@@ -288,6 +291,17 @@ pub struct InnerContext {
 
     /// True if account has subscribed to push notifications via IMAP.
     pub(crate) push_subscribed: AtomicBool,
+
+    /// [MagicEndpoint] needed for iroh peer channels.
+    /// If iroh_gossip lock is also needed, it should be aquired first.
+    pub(crate) iroh_endpoint: RwLock<Option<MagicEndpoint>>,
```

Comment:
A lot of the code could be simplified by using sth like https://docs.rs/async-once-cell/latest/async_once_cell/struct.OnceCell.html to do the lazy init, rather than using `RwLock<Option>`

Reply:
True, I don't know how I missed that..